src/Tasks.UnitTests/Exec_Tests.cs

```diff
@@ -48,6 +48,38 @@ private ExecWrapper PrepareExecWrapper(string command)
             return exec;
         }
 
+        [Fact]
+        public void EscapeParenthesesInPathToGeneratedBatchFile()
+        {
+            using (var testEnvironment = TestEnvironment.Create())
+            {
+                // This test counts files in TEMP. If it uses the system TEMP, some
+                // other process may interfere. Use a private TEMP instead.
+                var newTempPath = testEnvironment.CreateNewTempPathWithSubfolder("hello()wo(rld)").TempPath;
+
+                string tempPath = Path.GetTempPath();
+                Assert.StartsWith(newTempPath, tempPath);
+
+                // Get a count of how many temp files there are right now.
+                string[] tempFiles = Directory.GetFiles(tempPath);
```

Comment:
This is flaky, if the number of temp files in the directory changes during the run we'll fail and it'll be hard to investigate.

In general, I think we should have a single folder under TEMP, "MSBuild", and have everything under that folder. Maybe tests should use "MSBuildTest" to keep it separate. We should never enumerate the root of TEMP as it might be huge and some files might be locked, files are created and deleted and renamed there all the time.

Reply:
My question is why you're doing this in the first place. We care about Exec succeeding and producing the correct output, but I wouldn't bother with verifying we don't have garbage lying around—that has nothing to do with your change.

Reply:
This was leftover logic I forgot to get rid of.